com.microsoft.mrtk.spatialmanipulation/Solvers/Solver.cs

```diff
@@ -244,12 +244,9 @@ protected virtual void Start()
         {
             // Find the controller lookup class in the hierarchy. Solvers that require access to the
             // left, right or gaze controllers will use the references stored in this class.
-            //
-            // While FindObjectsOfType can be an expensive method, the controllerLookup variable is
-            // static. The hierarchy traversal will occur a maximum of once per scene.
             if (controllerLookup == null)
             {
-                ControllerLookup[] lookups = GameObject.FindObjectsOfType(typeof(ControllerLookup)) as ControllerLookup[];
+                ControllerLookup[] lookups = ComponentCache<ControllerLookup>.FindInstanceList();
```

Comment:
Can we change this to be FindFirstInstance() instead of the list. Allocating an array just to provide a warning seems like a waste of resources.

Reply:
If we wanted to provide an warning this could be localized to the `ControllerLookup` script file.  Maybe

```
        private void OnValidate()
        {
            if (FindObjectsByType<ControllerLookup>().Length > 1)
            {
                Debug.LogWarning("Found more than one instance of the ControllerLookup class in the hierarchy. There should only be one");
            }
        }
```

Or if we wanted to catch this outside of the Editor

```
        private void Start()
        {
            if (FindObjectsByType<ControllerLookup>().Length > 1)
            {
                Debug.LogWarning("Found more than one instance of the ControllerLookup class in the hierarchy. There should only be one");
            }
        }
```